Source/WebCore/Modules/mediastream/VideoTrackGenerator.cpp

```diff
@@ -122,13 +125,34 @@ Ref<MediaStreamTrack> VideoTrackGenerator::track()
     return Ref { m_track };
 }
 
-VideoTrackGenerator::Source::Source()
+VideoTrackGenerator::Source::Source(ScriptExecutionContextIdentifier identifier)
     : RealtimeMediaSource(CaptureDevice { { }, CaptureDevice::DeviceType::Camera, emptyString() })
+    , m_contextIdentifier(identifier)
 {
 }
 
+void VideoTrackGenerator::Source::endProducingData()
+{
+    ASSERT(isMainThread());
+    ScriptExecutionContext::postTaskTo(m_contextIdentifier, [weakThis = ThreadSafeWeakPtr { *this }] (auto&) {
+        auto protectedSource = weakThis.get();
```

Comment:
`RefPtr protectedSource`

Reply:
get already returns a RefPtr, but yeah, this is almost coding style now